src/stability_sdk/client.py

```diff
@@ -104,30 +119,147 @@ def process_artifacts_from_answers(
             idx += 1
 
 
-def open_channel(host: str, api_key: str = None) -> generation_grpc.GenerationServiceStub:
-    print(f"Opening channel {host}")
-    if host.endswith(":443"):
-        call_credentials = [grpc.access_token_call_credentials(api_key)]
-        channel_credentials = grpc.composite_channel_credentials(
-            grpc.ssl_channel_credentials(), *call_credentials
-        )
-        channel = grpc.secure_channel(host, channel_credentials)
-    else:
-        channel = grpc.insecure_channel(host)
-    return generation_grpc.GenerationServiceStub(channel)
-
 
 class ApiEndpoint:
     def __init__(self, stub, engine_id):
         self.stub = stub
         self.engine_id = engine_id
 
+class Project():
+    def __init__(self, api: 'Api', project: project.Project):
+        self.api = api
+        self.id = project.id
```

Comment:
recommend attaching the project object to the class as a hidden attribute and exposing these as properties. i.e.:

```python
class Project():
    def __init__(self, api: 'Api', project: project.Project):
        self.api = api
        #self.id = project.id
        self._project = project
        ...
    @property
    def id(self):
        return self._project.id
```

Reply:
I like it, will do!